### PR TITLE
RavenDB-13434: Added SearchOperator parameter to IRavenQueryable's Se…

### PR DESCRIPTION
**Pull request by maximburyak**
…arch

### CODE REVIEW

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs [hunk 1]
```diff
@@ -1322,19 +1322,27 @@ private void VisitSearch(MethodCallExpression searchExpression)
                 var expressionInfo = GetMember(expression.Arguments[1]);
                 if (LinqPathProvider.GetValueFromExpressionWithoutConversion(expression.Arguments[2], out value) == false)
                 {
-                    throw new InvalidOperationException("Could not extract value from " + expression);
+                    throw new InvalidOperationException("Could not extract searchTemrs value from " + expression);
```
Comment (ayende) [inline src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs]:
```suggestion
                    throw new InvalidOperationException("Could not extract searchTerms value from " + expression);
```

File: src/Raven.Client/Documents/LinqExtensions.cs [hunk 2]
```diff
@@ -829,27 +830,39 @@ public static Task<TSource> SingleOrDefaultAsync<TSource>(this IQueryable<TSourc
...
         /// </summary>
+        /// <typeparam name="T">The type of element of self</typeparam>
+        /// <param name="self">The <see cref="IQueryable{T}"/> to search on</param>
+        /// <param name="fieldSelector">Function returning the field to search on</param>
+        /// <param name="searchTerms">Field terms to search for, seperated with whitespaces</param>
+        /// <param name="boost">Boost factor for sorting purposes</param>
+        /// <param name="options">Logical operator to use in relation to the previous filtering statement.</param>
+        /// <param name="termsSearchOperator">Determines the logical operator between all of the terms received in searchTemrs parameter</param>
+        /// <returns></returns>
         public static IRavenQueryable<T> Search<T>(this IQueryable<T> self, Expression<Func<T, object>> fieldSelector, string searchTerms,
```
Comment (ayende) [inline src/Raven.Client/Documents/LinqExtensions.cs]: You need to create an overload here, not modify the method.
This is a backward compatability concern

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by ppekrol** [comment]
I've pushed some small adjustments. If this will go to 4.2 and as far as I understood this was the target, then we will not have any backward compatibility concerns here.

**Comment by ppekrol** [comment]
4.2 PR: https://github.com/ravendb/ravendb/pull/9176

**Comment by ayende** [comment]
This is important in v4.1, we have a customer that is being blocked by this.

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
Merged.

**Pull request closed without merging**